packages/flutter/lib/src/services/binary_messenger.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+/// A messenger which sends binary data across the Flutter platform barrier.
+///
+/// This class also registers handlers for incoming messages.
+abstract class BinaryMessenger {
+  /// A const constructor to allow subclasses to be const.
+  const BinaryMessenger();
+
+  /// Send a binary message to the platform plugins on the given channel.
+  ///
+  /// Returns a [Future] which completes to the received response, undecoded,
+  /// in binary form.
+  Future<ByteData> send(String channel, ByteData message);
+
+  /// Set a callback for receiving messages from the platform plugins on the
+  /// given channel, without decoding them.
+  ///
+  /// The given callback will replace the currently registered callback for that
+  /// channel, if any. To remove the handler, pass null as the [handler]
+  /// argument.
+  ///
+  /// The handler's return value, if non-null, is sent as a response, unencoded.
+  void setMessageHandler(String channel, Future<ByteData> handler(ByteData message));
+
+  /// Set a mock callback for intercepting messages from the `send*` methods on
```

Comment:
There's only one `send*` method on this class, so I think the dartdoc can be more specific.

Reply:
Done